File: Python Programming/problems on bitwise operator/problem19.py
'''
19. Write a program which accept one number from user and range of positions from user. Toggle all bits from that range. 
	
	Input : 897  9 13 
	Toggle all bits from position 9 to 13 of input number ie 7286.	
	
	Author : Milind Nalawade
	
	Date : 04/08/2021
	
	Language : Python 3.9.0

'''

import sys
#====
#Logic
#====
def ChkBit(no,ino1,ino2):
	ino=no^((1<<ino2)-ino1)
	print("\n ",ino)
#=============
#Entry point function
#=============
def main():
	no=int(sys.argv[1])
	ino1=int(sys.argv[2])
	ino2=int(sys.argv[3])
	ChkBit(no,ino1,ino2)
		
#=====
#starter
#=====
if __name__=="__main__":
	main()

File: Java Programming/problems on bitwise operator/problem19.java
/*
19. Write a program which accept one number from user and range of positions from user. Toggle all bits from that range. 
	
	Input : 897  9 13 
	Toggle all bits from position 9 to 13 of input number ie 7286.	
	
	Author : Milind Nalawade
	
	Date : 04/08/2021
	
	Language : Java 14.0.2
		
*/

import java.lang.*;

class Logic
{
	static void ChkBit(int no,int first,int last)
	{
		int ino=no^((1<<last)-first);
		System.out.print(" "+ino);
	}


}

class problem19 extends Logic
{
	public static void main(String args[])
	{
		int no=Integer.parseInt(args[0]);
		int ino1=Integer.parseInt(args[1]);
		int ino2=Integer.parseInt(args[2]);
		ChkBit(no,ino1,ino2);
	}
}